src/sql/util.py

```diff
@@ -559,6 +559,7 @@ def is_non_sqlalchemy_error(error):
         # Pyspark
         "UNRESOLVED_ROUTINE",
         "PARSE_SYNTAX_ERROR",
+        "AnalysisException",
```

Comment:
After looking through the code I think adding AnalysisException here will solve the issue since `PARSE_SYNTAX_ERROR` works as expected.
AnalysisException covers [all these error conditions](https://spark.apache.org/docs/latest/sql-error-conditions.html).

Reply:
I just need to test it somehow. Will try to package the jupysql and install it in a spark environment

Reply:
you can install like this:

```
pip install git+https://github.com/b1ackout/jupysql@running-sql-using-sparkconnect-should-not-print-full-stack-trace
```

Reply:
I tested it but no luck, the error message doesn't contain `AnalysisException`, only the sql error conditions listed [here](https://spark.apache.org/docs/latest/sql-error-conditions.html) which are included in AnalysisException. So instead of included all these in the list (which could also be updated regularly) I think checking if the error is of instance of AnalysisException would be a lot cleaner